[mohammedhamdhan] iP

mohammedhamdhan opened the pull request.

mohammedhamdhan committed:
Create new logo and edit greeting format

mohammedhamdhan committed:
Update .gitignore

Update .gitignore to ignore .class files

mohammedhamdhan committed:
Add functionality where Orca repeats input.
Echoes input and exits when "bye" is entered

mohammedhamdhan committed:
Merge branch 'master' of https://github.com/mohammedhamdhan/ip

mohammedhamdhan committed:
Add new ToDoList class

Add functionality to store entries and recite them when asked

mohammedhamdhan committed:
Add mark/unmark feature

Add new class TaskManager to handle all feature functionality of chatbot.
Add mark/unmark feature to chatbot.
Make code neater and more organized.

mohammedhamdhan committed:
Change class name to match A-CodingStandard

Lyam-T commented on src/main/java/Orca.java:
```diff
@@ -0,0 +1,48 @@
+import java.util.Scanner;
+
+public class Orca {
+    public static final String LINE = "----------------------------------------";
```
Should this be renamed to something more specific?

lishunyang12 replied:
I think they should be named as partitionLine instead, which seems to be more descriptive. Just a suggestion

Lyam-T commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,73 @@
+    private final Task[] tasks;
+    private int taskCount;
+
+    public TaskManager(int maxSize) {
```
Do you meant to make maxSize to be a constant? If so should it be capital instead?

lishunyang12 replied:
Constant names must be all uppercase using underscore to separate words. I think it should be MAXSIZE instead.

mohammedhamdhan committed:
Add new classes Todo, Deadline and Event.

Categorize task handling and organised code.

mohammedhamdhan committed:
Add try-catch statements

Expand error handling capabilities to handle more exceptions

mohammedhamdhan committed:
Add more catch sattements to handle exceptions

Add exception handling in events and deadlines, to ensure user input format is correct

mohammedhamdhan committed:
Categorize files into packages

mohammedhamdhan committed:
Edit formatting

mohammedhamdhan committed:
add delete feature

mohammedhamdhan committed:
add save feature

mohammedhamdhan committed:
Merge branch 'branch-Level-7'

aditichadha1310 commented on src/main/java/Orca/Storage.java:
```diff
@@ -0,0 +1,133 @@
+import java.util.Scanner;
+
+public class Storage {
+    // Use OS-independent file separators.
```
These comments seem to be for your reference instead of the reader. You may want to erase them

aditichadha1310 commented on src/main/java/Orca/Storage.java:
```diff
@@ -0,0 +1,133 @@
+        }
+
+        try (Scanner scanner = new Scanner(file)) {
+            while (scanner.hasNextLine()) {
```
Avoid the arrow-head style code.

aditichadha1310 commented on src/main/java/Orca/TaskManager.java:
```diff
@@ -0,0 +1,251 @@
+        }
+    }
+
+    public void addEventTask(String taskDescription) {
```
try to keep every method to less than equals to 30 lines of code. Avoid very long methods

mohammedhamdhan committed:
Extract classes: Ui, Parser, TaskList

mohammedhamdhan committed:
Add find feature

mohammedhamdhan committed:
Add JavaDoc comments

mohammedhamdhan committed:
Merge pull request #1 from mohammedhamdhan/branch-level-9

Add find feature

mohammedhamdhan committed:
Merge pull request #2 from mohammedhamdhan/branch-JavaDoc

Add JavaDoc comments

mohammedhamdhan committed:
Initial docs change

mohammedhamdhan committed:
Add formatting to docs

mohammedhamdhan committed:
Edit docs formatting
